### PR TITLE
Hide response buttons for past requests

### LINKED ISSUE #4770: hide accept decline buttons when request is in the past
Subject: hide accept decline buttons when request is in the past
Description:
self-explanatory

Results:
aapeli made a mistake here

Backend version: develop-65563756
Frontend version: develop-65563756
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36
Page: https://couchers.org/messages/request/81115
User: slabbster (4334)

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #4770.

**Web frontend checklist**
- [ ] Formatted my code with `yarn format`
- [ ] There are no warnings from `yarn lint --fix`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

File: app/web/features/messages/requests/HostRequestSendField.tsx [hunk 1]
```diff
@@ -175,99 +188,85 @@ export default function HostRequestSendField({
...
-                passHref
-              >
-                <StyledButton color="primary" component="a">
-                  {t("messages:write_reference_button_text")}
-                </StyledButton>
-              </Link>
-            )}
-          </>
+        {!isPast &&
+          (isHost ? (
```
Comment (nabramow) [inline app/web/features/messages/requests/HostRequestSendField.tsx]: Can you turn `!isPast && isHost` into a variable that's like `const showResponseButton` (or whatever it is) and use that instead? Just to make it a bit cleaner as it's already an unruly component.

Comment (aapeliv) [reply app/web/features/messages/requests/HostRequestSendField.tsx]: The `!isPast` applies to both the `isHost` and `!isHost` below... I could move them into another function/component in this file though to make it less crazy. Agree it's unwieldly.

**Review by nabramow — DISMISSED** [review]
Just that minor style comment about the variable but will tentatively approve as it works well.

**Comment by aapeliv** [comment]
@nabramow: I moved a bunch of stuff to smaller components to clean it up, let me know if you like it?

File: app/web/features/messages/requests/HostRequestSendField.tsx [hunk 2]
```diff
@@ -72,6 +72,153 @@ const StyledHelpTextContainer = styled("div")(({ theme }) => ({
...
+}: {
+  isPast: boolean;
+  isHost: boolean;
+  status: HostRequestStatus;
+}) {
+  const isHostPending =
+    !isPast &&
+    isHost &&
+    status === HostRequestStatus.HOST_REQUEST_STATUS_PENDING;
+
```
Comment (nabramow) [inline app/web/features/messages/requests/HostRequestSendField.tsx]: Would you mind moving these to separate components and importing them in here? That's generally the "React way". Also then you can just see an overview of what's happening in the parent component and click through to the children if needed. And then maybe you want to adjust the names `HostRequestRespondButtons` and `HostRequestGuideLinks` or whatever.

Looks waaaay better though!

Comment (aapeliv) [reply app/web/features/messages/requests/HostRequestSendField.tsx]: Will do

File: app/web/features/messages/requests/HostRequestSendField.tsx [hunk 3]
```diff
@@ -72,6 +72,153 @@ const StyledHelpTextContainer = styled("div")(({ theme }) => ({
...
+                Read our guide
+              </StyledLink>{" "}
+              on how to write a request that will get accepted.
+            </Trans>
+          </Typography>
+        </StyledHelpTextContainer>
+      )}
+    </>
+  );
+}
```
Comment (nabramow) [inline app/web/features/messages/requests/HostRequestSendField.tsx]: You probably want a null case at the top, like:

```javascript
if (!isHostPending && !isSurferRejected) {
    return null;
}
```

File: app/web/features/messages/requests/HostRequestSendField.tsx (hunk 2 above)
Comment (aapeliv) [reply app/web/features/messages/requests/HostRequestSendField.tsx]: Done!

File: app/web/features/messages/requests/HostRequestSendField.tsx (hunk 3 above)
Comment (aapeliv) [reply app/web/features/messages/requests/HostRequestSendField.tsx]: And done :)

**Review by nabramow — APPROVED** [review]
Looks good to me, so much more readable now thanks for breaking those components up!

**Pull request merged**